specviz/plugins/model_editor/items.py

```diff
@@ -24,38 +16,15 @@ def flux(self):
 
         result = self.model_editor_model.evaluate()
 
-        flux_units = self.data(self.DataRole).flux.unit
-
-        if self._plot_data_item is None:
-            model_flux_units = flux_units
-        else:
-            model_flux_units = u.Unit(self._plot_data_item.data_unit)
-
         if result is not None:
-            model_flux = result(self.model_spectral_axis.value) * model_flux_units
-            return model_flux.to(flux_units,
-                                 equivalencies=u.equivalencies.spectral_density(
-                                     self.model_spectral_axis))
-
-        return np.zeros(self.spectral_axis.size) * model_flux_units
+            return result(self.spectral_axis.value) * self.data(self.DataRole).flux.unit
 
-    @property
-    def model_spectral_axis(self):
-        if self._plot_data_item is None:
-            model_spectral_units = self.data(self.DataRole).spectral_axis.unit
-        else:
-            model_spectral_units = u.Unit(self._plot_data_item.spectral_axis_unit)
-        return self.spectral_axis.to(model_spectral_units,
-                                     equivalencies=u.equivalencies.spectral())
+        return np.zeros(self.spectral_axis.size) * self.data(self.DataRole).flux.unit
 
     @property
     def model_editor_model(self):
         return self._model_editor_model
 
     @model_editor_model.setter
     def model_editor_model(self, value):
-        self._model_editor_model = value
-
-    @property
-    def spectrum(self):
```

Comment:
Without the spectrum property, the arithmetics editor will receive the default Spectrum1D object living in this data item. That Spectrum1D has all its flux values set to 0. Maybe its best to update the data items Spectrum1D object each time the fitting params change? That might be a better way of handling the units as well.

Reply:
Yeah, as I mentioned below, it's just a matter of updating the `_data` attribute of the stored `Spectrum1D` object.